### ISSUE TITLE
Question: Pre-Filter Entries

### ISSUE THREAD

**Issue by animaux** [hananils/association_ui_selector#46]
Is there a way to add a filter to the possible selectable linked entries. E. g. only show entries with value X in field Y?

**Comment by nitriques** [comment]
I do not think so.

**Comment by animaux** [comment]
Hmnn, I wonder what kind of dirty hack would be needed to achieve this? :)

**Comment by nitriques** [comment]
If you want it to be performant, you need to do it in SQL. Filterings things in php does not scale.

Should not be that hard...

**Comment by animaux** [comment]
In my particular case performance would not be an issue.

Would it be conceivable that the UI Selector could use standard Symphony backend filter syntax? e. g. `?filter[bereich]=querwege`

**Comment by nitriques** [comment]
> Would it be conceivable that the UI Selector could use standard Symphony backend filter syntax?

Yes, but it uses SQL ;)

**Comment by animaux** [comment]
Hmn, ok. The most elegant solution of course would be a field setting to filter by a certain field/value …

**Comment by nitriques** [comment]
yup.

**Comment by animaux** [comment]
self::$provides = array(
            'association-ui' => array(
                'aui-selector' => 'Selector: Search',
                'aui-selector' => 'Selector: Search (pre-filtered)',
                'aui-selector-sortable' => 'Selector: Search (sortable)',
                'aui-selector-sortable' => 'Selector: Search (sortable, pre-filtered)',
            )
        );

**Comment by animaux** [comment]
(I wish it was that easy)

**Comment by nitriques** [comment]
LOL

**Comment by animaux** [comment]
Is @jonmifsud’s [filters](https://github.com/jonmifsud/association_ui_selector/tree/filters)-branch/fork offering something along these lines?

**Comment by jonmifsud** [comment]
Yes more or less, but it also comes with some required core changes.
Unfortunately I'm overloaded at the moment so it's a bit tough to get time
to clean it up by end of this month, but if you want to try put the
required pieces together it should work.

I have a site going live with these changes in around a week. There might
be other extensions eventually to compliment this functionality, with
filters coming up depending on changing values within an entry for example.
But can't promise a delivery date.

**Comment by animaux** [comment]
Thanks Jonathan! I guess the basic filtering I’m after wouldn’t need any core changes?

**Comment by jonmifsud** [comment]
It would for the simple reason that the Associations UI (when creating a
field) is part of the Core. If you look it up I have a PR against
integration with the changes. It works but some changes were requested,
mainly related to formatting.

All the UI does is take the options you save from the Core, and use them
for filtering.

**Comment by animaux** [comment]
Understand. So a dirty hack could be to use JS to filter the results further.

**Comment by jonmifsud** [comment]
Yes I guess so, my branch basically uses a 'data-filter' attribute.
Depending in what's inside that data item it will 'search'. There's also a
function you can call to clear up the results - useful when you change
filters via js. Not sure what's your timeline on this but if you need help
setting up let me know and I'll see if I can help.

**Comment by animaux** [comment]
Thanks Jon, it’s actually a nice-to-have-feature for site im currently working on.

**Comment by animaux** [comment]
Dear @jonmifsud,

has your code already wandered into the official codebase? Any news on this?

Cheers,

Alex

**Comment by jonmifsud** [comment]
Probably not - I'm abroad for the week so I doubt I'll be able to look into
it. will try push for it to be merged once back next week.

**Comment by animaux** [comment]
Ok, thanks!

**Comment by animaux** [comment]
I just browsed a bit through the code. Is this going to be in symphony 3 only? Was hoping for 2.6.x in fear of compatibility issues with other extensions.

**Comment by jonmifsud** [comment]
I think it would be a 3.x change - but if coupled with another extension /
custom configuration might be doable with 2.6.x if it's important I can try
to look into it in the coming weeks

**Comment by animaux** [comment]
Many thanks Jon. It’s *kind* of important. The site runs well like it is. It’s more a matter of making some parts of the backend easier to use and failsafe. Just let me know if you can look into it.
